pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,337 @@
+		input:
+			bam_file = UMIAwareDuplicateMarking.duplicate_marked_bam,
+			genes_gtf = gtf,
+			sample_id = GetSampleName.sample_name
+	}
+
+	# This should be removed before turning this into production
+	call rnaseqc2 as RNASeQC2Transcriptome {
+		input:
+			bam_file = UMIAwareDuplicateMarkingTranscriptome.duplicate_marked_bam,
+			genes_gtf = gtf,
+			sample_id = GetSampleName.sample_name
+	}
+
+	call CollectRNASeqMetrics {
+		input:
+			input_bam=UMIAwareDuplicateMarking.duplicate_marked_bam,
+			input_bam_index=UMIAwareDuplicateMarking.duplicate_marked_bam_index,
+			output_bam_prefix=GetSampleName.sample_name,
+			ref_dict=refDict,
+			ref_fasta=ref,
+			ref_fasta_index=refIndex,
+			ref_flat=refFlat,
+			ribosomal_intervals=ribosomalIntervals,
+			preemptible_tries=0
+	}
+
+	call CollectMultipleMetrics {
+		input:
+			input_bam=UMIAwareDuplicateMarking.duplicate_marked_bam,
+			input_bam_index=UMIAwareDuplicateMarking.duplicate_marked_bam_index,
+			output_bam_prefix=GetSampleName.sample_name,
+			ref_dict=refDict,
+			ref_fasta=ref,
+			ref_fasta_index=refIndex,
+			preemptible_tries=0
+	}
+
+
+  output {
+	File transcriptome_bam = UMIAwareDuplicateMarkingTranscriptome.duplicate_marked_bam
+	File transcriptome_bam_index = UMIAwareDuplicateMarkingTranscriptome.duplicate_marked_bam_index
+	File transcriptome_duplicate_metrics = UMIAwareDuplicateMarkingTranscriptome.duplicate_metrics
+	File output_bam = UMIAwareDuplicateMarking.duplicate_marked_bam
+	File output_bam_index = UMIAwareDuplicateMarking.duplicate_marked_bam_index
+	File duplicate_metrics = UMIAwareDuplicateMarking.duplicate_metrics
+	File gene_tpm = rnaseqc2.gene_tpm
+	File gene_counts = rnaseqc2.gene_counts
+	File exon_counts = rnaseqc2.exon_counts
+	File metrics = rnaseqc2.metrics
+		
+  }
+}
+
+task STAR {
+	input {
+		File bam
+		File starIndex
+	}
+	Int disk_space = ceil(2.2 * size(bam, "GB") + size(starIndex, "GB")) + 250
+
+	command <<<
+		echo $(date +"[%b %d %H:%M:%S] Extracting STAR index")
+		mkdir star_index
+		tar -xvvf ~{starIndex} -C star_index --strip-components=1
+
+		STAR --readFilesIn ~{bam} --readFilesType SAM PE --readFilesCommand samtools view -h \
+			--runMode alignReads --genomeDir star_index --outSAMtype BAM Unsorted --runThreadN 8 \
+			--limitSjdbInsertNsj 1200000 --outSAMstrandField intronMotif --outSAMunmapped Within \
+			--outFilterType BySJout --outFilterMultimapNmax 20 --outFilterScoreMinOverLread 0.33 \
+			--outFilterMatchNminOverLread 0.33 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.1 \
+			--alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 \
+			--alignSJDBoverhangMin 1 --alignSoftClipAtReferenceEnds Yes --chimSegmentMin 15 --chimMainSegmentMultNmax 1 \
+			--chimOutType WithinBAM SoftClip --chimOutJunctionFormat 0 --twopassMode Basic --quantMode TranscriptomeSAM --quantTranscriptomeBan Singleend
+
+	>>>
+
+	runtime {
+		docker : "us.gcr.io/tag-team-160914/neovax-tag-rnaseq:v1"
```

Comment:
Can you make this docker publicly available - or use the gtex docker at: gcr.io/broad-cga-francois-gtex/gtex_rnaseq:V9 or add the Dockerfile to this PR?

Reply:
I think Micah gave us access to us.gcr.io/tag-team-160914, let me know if the issue persists.

Reply:
Unfortunately our test framework doesn't run as me / so I need to figure out how to access that docker from the test runner (which runs on the gotc-dev-cromwell instance) or, better, but this docker in a publicly accessible location.

Reply:
If we could get access to the Dockerfile and associated scripts I'd be happy to migrate it over to WARP...ideally we would like to do this for all of the 'custom' docker images used in this pipeline that are not already in WARP

Reply:
The docker that is being used here also has samtools built into it, so we were unable to use the docker that you referenced @kcibul , there's a ticket on the board: https://broadinstitute.atlassian.net/browse/GL-1870 to cover breaking this task up into separate samtools and star docker tasks (and presumably use the STAR version you reference).